frontend/src/components/DisconnectPeerDialog.tsx

```diff
@@ -0,0 +1,66 @@
+import { toast } from "src/components/ui/use-toast";
+import { usePeers } from "src/hooks/usePeers";
+import { Peer } from "src/types";
+import { request } from "src/utils/request";
+import {
+  AlertDialogAction,
+  AlertDialogCancel,
+  AlertDialogContent,
+  AlertDialogDescription,
+  AlertDialogFooter,
+  AlertDialogHeader,
+  AlertDialogTitle,
+} from "./ui/alert-dialog";
+
+type Props = {
+  peer: Peer;
+};
+
+export function DisconnectPeerDialog({ peer }: Props) {
+  const { mutate: reloadPeers } = usePeers();
+
+  async function disconnectPeer() {
+    try {
+      if (!peer.nodeId) {
```

Comment:
how can a peer exist but not the nodeId?

Reply:
@im-adithya can you please test this? otherwise we are adding extra unnecessary code and setting up poor conventions which will be copied through the app

Reply:
I had the same question, but this check was used previously so I assumed there might be an issue with one of the backends where nodeId is not returned. Should I remove this?